docs/backend/schema.ddl

```diff
@@ -0,0 +1,281 @@
+}
+
+Table WorkflowIntermediateEntities {
+  entity_id INT [not null, ref: > Entity.id]
+  workflow_id INT [not null, ref: > Workflow.id]
+
+  Indexes {
+    (entity_id, workflow_id) [unique]
+  }
+}
+
+Table WorkflowOutputs {
+  entity_id INT [not null, ref: > Entity.id]
+  workflow_id INT [not null, ref: > Workflow.id]
+
+  Indexes {
+    (entity_id, workflow_id) [unique]
+  }
+}
+
+
+///////////////////////////////////////////////////
+// Public accession ids
+
+Table PublicRepository {
+  id INT [pk, increment]
+  name VARCHAR [not null, unique]
+  website VARCHAR [unique]
+}
+
+Table Accession {
+  entity_id INT [not null, ref: > Entity.id]
+  public_repository_id INT [not null, ref: > PublicRepository.id]
+
+  Indexes {
+    (entity_id, public_repository_id) [unique]
+  }
+}
+
+
+///////////////////////////////////////////////////
+// Sequences
+
+// describes a sequencer machine  (TODO: what's a good way of canonicalizing this?)
+Table Sequencer {
+  id INT [pk, increment]
+  name VARCHAR [not null, unique]
+}
+
+// describes a sequencing protocol  (TODO: what's a good way of canonicalizing this?)
+Table SequencingProtocol {
+  id INT [pk, increment]
+  name VARCHAR [not null, unique]
+}
+
+// describes a single set of sequence data from the sequencer. (FASTQ)
+Table RawSequence {
+  id INT [pk, increment]
+  entity_id INT [not null, unique, ref: > Entity.id]
+
+  physical_sample_id INT [not null, ref: > PhysicalSample.id]
+  s3_bucket VARCHAR [not null]
+  s3_key VARCHAR [not null, unique]
+
+  sequencer_id INT [not null, ref: > Sequencer.id]
+  sequencing_protocol_id INT [not null, ref: > SequencingProtocol.id]
+
+  // sequencing date is optional, but sample collection date should not be.
+  sequencing_date FLOAT
+
+  // optional field for gisaid submission, we'd like to get users to provide this.
+  sequencing_depth FLOAT
+
+  Indexes {
+    (s3_bucket, s3_key) [unique]
+  }
+}
+
+// describes a single pathogen sequence
+Table PathogenSequence {
+  id INT [pk, increment]
+  entity_id INT [not null, unique, ref: > Entity.id]
+
+  sequence VARCHAR [not null]
+
+  // sequence statistics we can calculate from the fasta file
+  num_unambiguous_sites INT [not null]
+  num_n INT [not null]
+  num_mixed INT [not null]
+
+}
+
+// describes a single sequence that was uploaded to the system directly.
+Table UploadedPathogenSequence {
+  pathogen_sequence_id INT [not null, unique, ref: > PathogenSequence.id]
+  physical_sample_id INT [not null, ref: > PhysicalSample.id]
+
+  // optional field for gisaid submission, we'd like to get users to provide this.
+  sequencing_depth FLOAT
+}
+
+// describes a single sequence that was called from a consensus genome run
+Table CalledPathogenSequence {
+  pathogen_sequence_id INT [not null, unique, ref: > PathogenSequence.id]
+}
+
+///////////////////////////////////////////////////
+// Consensus genome pipeline
+
+// TODO: @jackkamm do the BAM/SRA records need to store any metrics?
+
+Table HostFilteredBam {
```

Comment:
`HostFiltered` is redundant since it's aligned to SARS-CoV-2. Just call it `Bam` or `AlignedReads`.

Also, this needs to be linked to the `RawSequence` and the `CalledPathogenSequence` somehow

Reply:
Add a column for the reference genome accession

Reply:
> `HostFiltered` is redundant since it's aligned to SARS-CoV-2. Just call it `Bam` or `AlignedReads`.

👍 

> Also, this needs to be linked to the `RawSequence` and the `CalledPathogenSequence` somehow

It's done through the workflow objects.

> Add a column for the reference genome accession

Anything that is an entity can have an accession (which you noted was missing).